### PR TITLE
SWITCHYARD-2955 - resolving facet warnings

### PR DESCRIPTION
**Pull request by bfitzpat**
- also includes a change for JBTIS-887 - updating to 4.4.0.Alpha2 TP for Neon
- note that this may fail at the CI level - I'm getting a weird timeout locally with the m2e tests

### CODE REVIEW

File: eclipse/plugins/org.switchyard.tools.ui/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -44,7 +44,11 @@ Require-Bundle: org.eclipse.ui,
  org.jboss.ide.eclipse.as.wtp.ui,
  org.fusesource.ide.server.karaf.core,
  org.eclipse.jdt.launching,
- org.jboss.ide.eclipse.as.wtp.core
+ org.jboss.ide.eclipse.as.wtp.core,
+ org.jboss.tools.maven.core,
```
Comment (bfitzpat) [inline eclipse/plugins/org.switchyard.tools.ui/META-INF/MANIFEST.MF]: These are the changes to avoid the facet warnings.

**Comment by bfitzpat** [comment]
@apupier The test now fails 100% of the time on a timeout when it tries to import the maven project. Not sure why. It worked fine before. But at least it's consistent. Let's take a look next week

`!ENTRY org.fusesource.ide.camel.model.service.core 2 0 2016-09-30 12:41:13.215
!MESSAGE Start search of Camel Java Route in project switchyard-bean-service
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 180.298 sec <<< FAILURE! - in org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest
testBeanServiceQuickstart(org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest)  Time elapsed: 178.961 sec  <<< FAILURE!
java.lang.AssertionError: Timeout while waiting for completion of job: Updating Maven Project(153)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.junit.Assert.assertFalse(Assert.java:64)
        at org.eclipse.m2e.tests.common.JobHelpers.waitForJobs(JobHelpers.java:145)
        at org.eclipse.m2e.tests.common.JobHelpers.waitForBuildJobs(JobHelpers.java:134)
        at org.eclipse.m2e.tests.common.JobHelpers.waitForJobsToComplete(JobHelpers.java:51)
        at org.eclipse.m2e.tests.common.AbstractMavenProjectTestCase.waitForJobsToComplete(AbstractMavenProjectTestCase.java:442)
        at org.eclipse.m2e.tests.common.AbstractMavenProjectTestCase.waitForJobsToComplete(AbstractMavenProjectTestCase.java:438)
        at org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest.runProjectTest(SwitchYardConfigurationTest.java:59)
        at org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest.testBeanServiceQuickstart(SwitchYardConfigurationTest.java:41)


Results :

Failed tests:
  Timeout while waiting for completion of job: Updating Maven Project(153)

Tests run: 1, Failures: 1, Errors: 0, Skipped: 0

`

**Comment by bfitzpat** [comment]
Ok. I've figured a few things out. The only one left I can't figure out is the org.fusesource.ide.projecttemplates one that exposes jst.camel. When I try to add it as a dependency, that's when things lock up on the m2e test.

Without it I'm down to just this warning:
```
!ENTRY org.eclipse.wst.common.project.facet.core 4 0 2016-10-03 14:22:13.797
!MESSAGE Project facet jst.camel has not been defined. It is used in plugin org.fusesource.ide.server.karaf.core.

```
I think maybe if we can move the facet definition out of projecttemplates and back into one of the core plug-ins that would work better.

WDYT @apupier ?

(same comment I put on the JIRA SWITCHYARD-2955)

**Comment by apupier** [comment]
I don't think that moving the jst.camel definition will change anything, it's just that when you don't add it, the camel facet i snot added and so it means that we have a deadlock when we try to install the Camel facet in your tests and that we need to fix that

**Comment by bfitzpat** [comment]
The deadlock happens on the import of the project. The test isn't doing anything special.

**Comment by apupier** [comment]
I wasn't precise precise enough, the test emphasize a potential deadlock which occurs everytime in this case, so we have a real issue to fix.

**Comment by bfitzpat** [comment]
Good point. Unfortunately I'm at an impasse. Not sure what else to investigate at this point. Any suggestions on how to proceed?

**Comment by bfitzpat** [comment]
Well, as I suspected - there may be some imports missing, which was causing the projecttemplates plug-in not to load properly. 

When I add these to the dependencies - 
'''
org.eclipse.wst.common.project.facet.core,
 org.eclipse.jst.common.project.facet.core,
 org.fusesource.ide.projecttemplates
'''

I get past the deadlock, but it's still not seeing the jst.camel facet.
'''
TEST-SETUP: testBeanServiceQuickstart
!SESSION 2016-10-04 12:05:25.952 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.8.0_45
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86, WS=win32, NL=en_US
Framework arguments:  -application org.eclipse.tycho.surefire.osgibooter.uitest -testproperties E:\Users\bfitzpat\github\tools (GitHub Desktop)\eclipse\tests\org.switchyard.tools.m2e.tests\target\surefire.properties -product org.eclipse.platform.ide -pluginCustomization E:\Users\bfitzpat\github\tools (GitHub Desktop)\eclipse\tests\org.switchyard.tools.m2e.tests/plugin_customization.ini
Command-line arguments:  -data E:\Users\bfitzpat\github\tools (GitHub Desktop)\eclipse\tests\org.switchyard.tools.m2e.tests\target\work\data -application org.eclipse.tycho.surefire.osgibooter.uitest -testproperties E:\Users\bfitzpat\github\tools (GitHub Desktop)\eclipse\tests\org.switchyard.tools.m2e.tests\target\surefire.properties -product org.eclipse.platform.ide -pluginCustomization E:\Users\bfitzpat\github\tools (GitHub Desktop)\eclipse\tests\org.switchyard.tools.m2e.tests/plugin_customization.ini -consoleLog

!ENTRY org.eclipse.wst.common.project.facet.core 4 0 2016-10-04 12:05:47.281
!MESSAGE Project facet jst.camel has not been defined. It is used in plugin org.fusesource.ide.server.karaf.core.

!ENTRY org.eclipse.wst.common.project.facet.core 4 0 2016-10-04 12:05:47.290
!MESSAGE Project facet jst.camel has not been defined. It is used in plugin org.fusesource.ide.server.fuse.core.
log4j:WARN No appenders could be found for logger (org.jboss.logging).
log4j:WARN Please initialize the log4j system properly.
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 29.077 sec - in org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest
testBeanServiceQuickstart(org.switchyard.tools.m2e.tests.SwitchYardConfigurationTest)  Time elapsed: 28.028 sec

Results :

Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
'''

**Comment by bfitzpat** [comment]
Other tests are still timing out later in the cycle.

**Comment by bfitzpat** [comment]
OK. I have it sort of working except the testCamelRouteValidation, which also fails here. The issue seems to be with resolving the camel version - which works fine when you import the project locally. 

```
12:29:19 TEST-SETUP: testCamelRouteValidation
12:29:25 Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 154.388 sec <<< FAILURE! - in org.switchyard.tools.ui.tests.SwitchYardValidatorTest
12:29:25 testWiringValidation(org.switchyard.tools.ui.tests.SwitchYardValidatorTest)  Time elapsed: 148.827 sec
12:29:25 testCamelRouteValidation(org.switchyard.tools.ui.tests.SwitchYardValidatorTest)  Time elapsed: 5.556 sec  <<< ERROR!

```

File: eclipse/plugins/org.switchyard.tools.m2e/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -12,5 +12,9 @@ Require-Bundle: org.eclipse.core.runtime;bundle-version="3.4.0",
  org.eclipse.wst.common.project.facet.core;bundle-version="1.4.0",
  org.eclipse.m2e.wtp;bundle-version="0.16.0",
  org.eclipse.jdt.core;bundle-version="3.3.0",
+ org.eclipse.wst.jsdt.web.core;visibility:=reexport,
+ org.eclipse.jpt.jpa.core;visibility:=reexport,
```
Comment (apupier) [inline eclipse/plugins/org.switchyard.tools.m2e/META-INF/MANIFEST.MF]: why do you need to reexport dependencies?

File: eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -26,6 +26,11 @@ Require-Bundle: org.eclipse.core.runtime;bundle-version="3.4.0",
  org.switchyard.tools.ui.bpel,
  org.switchyard.tools.ui.bpmn2,
  org.eclipse.m2e.core;bundle-version="[1.4.0,2.0.0)",
- org.eclipse.equinox.event
+ org.eclipse.equinox.event,
+ org.eclipse.wst.common.project.facet.core,
+ org.eclipse.jst.common.project.facet.core,
+ org.eclipse.wst.common.project.facet.ui,
+ org.eclipse.jst.common.project.facet.ui,
+ org.fusesource.ide.camel.model.service.v2_17_3;visibility:=reexport
```
Comment (apupier) [inline eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF]: why do you need to reexport here too?

File: eclipse/tests/org.switchyard.tools.ui.tests/settings.xml [hunk 4]
```diff
@@ -11,7 +11,8 @@
         <repository>
           <id>jboss-public-repository-group</id>
           <name>JBoss Public Maven Repository Group</name>
-          <url>https://repository.jboss.org/nexus/content/groups/public/</url>
+          <url>https://origin-repository.jboss.org/nexus/content/groups/public/</url>
+<!--           <url>https://repository.jboss.org/nexus/content/groups/public/</url> -->
```
Comment (apupier) [inline eclipse/tests/org.switchyard.tools.ui.tests/settings.xml]: i would recommend either to remove the commented line or to explain why it is commented

File: eclipse/tests/org.switchyard.tools.ui.tests/settings.xml [hunk 5]
```diff
@@ -41,7 +42,8 @@
         <pluginRepository>
           <id>jboss-public-repository-group</id>
           <name>JBoss Public Maven Repository Group</name>
-          <url>https://repository.jboss.org/nexus/content/groups/public/</url>
+          <url>https://origin-repository.jboss.org/nexus/content/groups/public/</url>
+<!--           <url>https://repository.jboss.org/nexus/content/groups/public/</url> -->
```
Comment (apupier) [inline eclipse/tests/org.switchyard.tools.ui.tests/settings.xml]: i would recommend either to remove the commented line or to explain why it is commented

File: eclipse/tests/org.switchyard.tools.ui.tests/src/org/switchyard/tools/ui/tests/AbstractSwitchYardTest.java [hunk 6]
```diff
@@ -78,4 +81,18 @@ public boolean matches(Job job) {
                     || job.getClass().getName().endsWith("JREUpdateJob"); //$NON-NLS-1$
         }
     }
+
+    protected void waitForJobs() throws OperationCanceledException, InterruptedException {
```
Comment (apupier) [inline eclipse/tests/org.switchyard.tools.ui.tests/src/org/switchyard/tools/ui/tests/AbstractSwitchYardTest.java]: FAR (For another review): we really need to put this method in a good place to factorize it

File: pom.xml [hunk 7]
```diff
@@ -15,7 +15,7 @@
   <properties>
     <!-- Just in case the Eclipse project version does not match the parent. -->
     <version.switchyard.runtime>2.1.0.Final</version.switchyard.runtime>
-    <jbtis.version>4.4.0.Alpha2-SNAPSHOT</jbtis.version>
+    <jbtis.version>4.4.0.Alpha2</jbtis.version>
     
     <!-- Tycho versions -->
     <jboss-tycho-version>0.23.1</jboss-tycho-version>
```
Comment (apupier) [inline pom.xml]: FAR: migth worth updating the tycho version to reduce technical debt. when we have build issue, teh first thing that people are asking is "which version o ftycho are you using?", 0.26 is available

File: eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF (hunk 3 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF]: I will try without the reexports

File: eclipse/tests/org.switchyard.tools.ui.tests/settings.xml (hunk 4 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/settings.xml]: ok

File: eclipse/tests/org.switchyard.tools.ui.tests/settings.xml (hunk 5 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/settings.xml]: ok

File: eclipse/tests/org.switchyard.tools.ui.tests/src/org/switchyard/tools/ui/tests/AbstractSwitchYardTest.java (hunk 6 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/src/org/switchyard/tools/ui/tests/AbstractSwitchYardTest.java]: agreed

File: pom.xml (hunk 7 above)
Comment (bfitzpat) [inline pom.xml]: ok. I'll check with Paul and we'll adjust accordingly

File: eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF (hunk 3 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/META-INF/MANIFEST.MF]: Works fine without the reexports. Removed them.

File: eclipse/tests/org.switchyard.tools.ui.tests/settings.xml (hunk 5 above)
Comment (bfitzpat) [inline eclipse/tests/org.switchyard.tools.ui.tests/settings.xml]: Removed the line.

File: pom.xml (hunk 7 above)
Comment (bfitzpat) [inline pom.xml]: Adjusted to 0.26 and all is well locally. Waiting for PR build to complete.

**Comment by bfitzpat** [comment]
Closing. Was done across a set of three PRs and fixed.

**Pull request closed without merging**